miden-tx/src/tests/kernel_tests/test_note.rs

```diff
@@ -520,3 +520,32 @@ fn test_get_inputs_hash() {
 
     assert_eq!(process.get_stack_state()[0..16], expected_stack);
 }
+
+#[test]
+fn test_get_current_script_hash() {
+    let tx_context = TransactionContextBuilder::with_standard_account(ONE)
+        .with_mock_notes_preserved()
+        .build();
+
+    // calling get_note_script_hash should return script hash
+    let code = "
+    use.kernel::prologue
+    use.kernel::note->note_internal
+    use.kernel::note
+
+    begin
+        exec.prologue::prepare_transaction
+        exec.note_internal::prepare_note
+        dropw dropw dropw dropw
+        exec.note::get_note_script_hash
+
+        # truncate the stack
+        swapw dropw
+    end
```

Comment:
You said that you used the test for the `get_serial_number` as a reference, but it turned out that this test has some flaws in it. Your test (and `get_serial_number` one) could be rewritten like so:
```masm
    use.kernel::prologue
    use.miden::note
    begin
        exec.prologue::prepare_transaction
        exec.note::get_note_script_hash

        # truncate the stack
        swapw dropw
    end
```

So there are three comments:
1. As far as I know, we should avoid using procedures from the `kernel` lib directly, so in that case we should use the `get_note_script_hash` procedure from the `miden` lib, and not from the `kernel`.
2. We don't need to import the same module with different names. Probably at first there were both `miden::note` and `kernel::note` modules, that's why we wanted to have `kernel::note` as an internal one, but for now it is unnecessary.
3. We don't need to call `note::prepare_note` procedure — it just moves some data on the stack, without updating any internal variables. We don't use this data, so this execution could be safely removed.

Reply:
Thank you for the explanation! I fixed this test and the serial number one.